### PR TITLE
Build: Update the project for compatibility with Node.js 10/12/14

### PR DESCRIPTION
**Pull request by mgol**
Ref gh-390

### CODE REVIEW

**Comment by fnagel** [comment]
I've tested the PR using a Docker image: https://github.com/fnagel/docker-jqueryui

Building custom downloads works. Using the Themeroller (GET variable loading for custom themes) does not work. I'm getting errors with some node modules (see screenshot). Does that work for you? 

grafik

File: package.json [hunk 1]
```diff
@@ -7,41 +7,42 @@
   },
   "license": "MIT",
   "dependencies": {
```
Comment (fnagel) [inline package.json:9]: On `npm install` I get this error but everything works anyway. Looks like my Docker image misses a dependency. Not relevant, isn't it?

```
> modern-syslog@1.2.0 install /home/node/app/node_modules/modern-syslog
> node-gyp rebuild

gyp ERR! build error 
gyp ERR! stack Error: not found: make
gyp ERR! stack     at getNotFoundError (/usr/local/lib/node_modules/npm/node_modules/which/which.js:13:12)
gyp ERR! stack     at F (/usr/local/lib/node_modules/npm/node_modules/which/which.js:68:19)
gyp ERR! stack     at E (/usr/local/lib/node_modules/npm/node_modules/which/which.js:80:29)
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/which.js:89:16
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/isexe/mode.js:8:5
gyp ERR! stack     at FSReqCallback.oncomplete (fs.js:181:21)
gyp ERR! System Linux 4.19.104-microsoft-standard
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/node/app/node_modules/modern-syslog
gyp ERR! node -v v14.13.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
```

File: package.json [hunk 2]
```diff
@@ -53,5 +53,8 @@
   ],
   "scripts": {
     "test": "nodeunit test"
+  },
+  "engines": {
+    "node": "10 || 12 || 14"
```
Comment (fnagel) [inline package.json:60]: I've tested with version 14 only.

File: external/farbtastic/README.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
```
Comment (fnagel) [inline external/farbtastic/README.html:1]: Shall we remove the readme.html and only keep the readme.md?

File: external/farbtastic/demo1.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
```
Comment (fnagel) [inline external/farbtastic/demo1.html:1]: Shall we remove the demo files?

File: .travis.yml [hunk 5]
```diff
@@ -1,11 +1,9 @@
...
 language: node_js
 node_js:
-  - "0.12"
+  - "10"
 addons:
   apt:
     packages:
     - xsltproc
 before_script:
-  - npm install -g grunt-cli
```
Comment (fnagel) [inline .travis.yml:10]: Is `grunt-cli` no longer needed?

**Review by fnagel — COMMENTED** [review]
Checked each commit. More for logic things than CGL or similar. Looks good!

File: .travis.yml (hunk 5 above)
Comment (mgol) [reply .travis.yml:10]: We have it in `devDependencies` which is a recommended practice as you don't depend on global packages that may change any moment.

File: external/farbtastic/README.html (hunk 3 above)
Comment (mgol) [reply external/farbtastic/README.html:1]: Sure, will do.

File: external/farbtastic/demo1.html (hunk 4 above)
Comment (mgol) [reply external/farbtastic/demo1.html:1]: Sure, will do.

File: package.json (hunk 1 above)
Comment (mgol) [reply package.json:9]: This package also doesn't install on jQuery infra so I made it optional in the PR and I fallback to regular `console` methods. You can see it's in `optionalDependencies`.

File: package.json (hunk 2 above)
Comment (mgol) [reply package.json:60]: Cool, I was mostly testing with Node 10. :)

File: package.json (hunk 1 above)
Comment (fnagel) [reply package.json:9]: Ahh ok. So this should be no problem.

File: .travis.yml (hunk 5 above)
Comment (fnagel) [reply .travis.yml:10]: Ahh ok. Something I've learned today ;-)

**Review by fnagel — APPROVED** [review]
Tested with node-14. Download builder works. Themeroller has issue with link generation, see https://github.com/jquery/download.jqueryui.com/pull/552#issuecomment-706056421 Needs to be fixed in the actual Themeroller repository.

**Pull request merged**